python/dolfin/io.py

```diff
@@ -1,19 +1,92 @@
 # -*- coding: utf-8 -*-
-# Copyright (C) 2017 Chris N. Richardson, Garth N. Wells, Michal Habera
+# Copyright (C) 2017-2018 Chris N. Richardson, Garth N. Wells and Michal Habera
 #
 # This file is part of DOLFIN (https://www.fenicsproject.org)
 #
 # SPDX-License-Identifier:    LGPL-3.0-or-later
-
-"""IO interfaces for inout data, post-processing and checkpointing"""
+"""IO module for input data, post-processing and checkpointing"""
 
 import dolfin.cpp as cpp
 from dolfin.function.function import Function
 
 
-class HDF5File(cpp.io.HDF5File):
+class HDF5File:
     """Interface to HDF5 files"""
 
+    def __init__(self, mpi_comm, filename: str, mode: str):
+        """Open HDF5 file
+
+        Parameters
+        ----------
+        mpi_comm
+            The MPI communicator
+        filename
+            Name of the file
+        mode
+            File opening mode, which can be write (w), read (r) or append (a)
+
+        """
+        self._cpp_object = cpp.io.HDF5File(mpi_comm, filename, mode)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        return self._cpp_object.close()
+
+    def get_mpi_atomicity(self) -> bool:
+        """Get atomicity of the HDF5 file"""
+        return self._cpp_object.get_mpi_atomicity()
+
+    def set_mpi_atomicity(self, atomicity: bool):
+        """Set atomicity of the HDF5 file"""
+        self._cpp_object.set_mpi_atomicity(atomicity)
+
+    def close(self):
+        """Close file"""
+        self._cpp_object.close()
+
+    def write(self, o, name, t=None):
+        """Write object to file"""
+        o_cpp = getattr(o, "_cpp_object", o)
+        if t is None:
+            self._cpp_object.write(o_cpp, name)
+        else:
+            self._cpp_object.write(o_cpp, name, t)
+
+    def read_mvc(self, mesh, name: str = ""):
+        # FIXME: figure type out from file (or pass string)  and return?
+        raise NotImplementedError("General MVC read function not implemented.")
```

Comment:
Should be easy to determine the type of a dataset in HDF5 with a small utility function.